### PR TITLE
many: go vet cleanups

### PR DESCRIPTION
**Pull request by bboozzoo**
Cleaning up most of `go vet` 1.10 issues. 

These 2 problems remain unresolved though:
```
# github.com/snapcore/snapd/overlord/ifacestate_test                 
overlord/ifacestate/ifacestate_test.go:120:29: internal compiler error: no function definition for [0xc421125440] FUNC-method(*ifacestate.InterfaceManager) func()                                                                            
                                                                            
                                                                                                                                                                
Please file a bug report including a short program that triggers the error.                                                         
https://golang.org/issue/new                                                                                                                                      
# github.com/snapcore/snapd/overlord/devicestate_test                                                                                                                                                                                         
overlord/devicestate/devicestate_test.go:403:38: internal compiler error: no function definition for [0xc420e1aba0] FUNC-method(*devicestate.DeviceManager) func() asserts.KeypairManager
                                                                                                                                   
                                                                                                                                                                                                                                              
Please file a bug report including a short program that triggers the error.                                                         
https://golang.org/issue/new                                                                                                                                                          
```

### CODE REVIEW

File: userd/ui/ui.go [hunk 1]
```diff
@@ -56,15 +56,15 @@ var hasKDialogExecutable = func() bool {
...
-	hasZenityExecutable := f
+	hasZenityExecutable = f
 	return func() {
 		hasZenityExecutable = oldHasZenityExecutable
 	}
 }
 
 func MockHasKDialogExecutable(f func() bool) func() {
 	oldHasKDialogExecutable := hasKDialogExecutable
-	hasKDialogExecutable := f
```
Comment (zyga) [inline userd/ui/ui.go]: Nice catch. Do we need this for 2.32?

File: userd/ui/ui.go [hunk 2]
```diff
@@ -56,15 +56,15 @@ var hasKDialogExecutable = func() bool {
 
 func MockHasZenityExecutable(f func() bool) func() {
 	oldHasZenityExecutable := hasZenityExecutable
-	hasZenityExecutable := f
+	hasZenityExecutable = f
```
Comment (zyga) [inline userd/ui/ui.go]: Same question as below

File: snap/info_test.go [hunk 3]
```diff
@@ -493,7 +493,7 @@ func ExampleSplitSnapApp() {
 	// Output: hello-world env
 }
 
-func ExampleSplitSnapAppShort() {
+func ExampleSplitSnapApp_short() {
```
Comment (zyga) [inline snap/info_test.go]: Hmm? Did go vet crash without this change?

File: userd/ui/ui.go (hunk 1 above)
Comment (stolowski) [reply userd/ui/ui.go]: ouch /o\. No, it's just affecting the tests.

File: snap/info_test.go (hunk 3 above)
Comment (bboozzoo) [reply snap/info_test.go]: This is what vet complained about:
`snap/info_test.go:496: ExampleSplitSnapAppShort refers to unknown identifier: SplitSnapAppShort`

Apparently this is a feature https://blog.golang.org/examples (although I'd say it's somewhat obscure)

**Comment by bboozzoo** [comment]
> LGTM but let’s please report an upstream bug as well.

I left a note in the original ticket https://github.com/golang/go/issues/23701#issuecomment-372995434

**Pull request merged**